oqupy/system.py

```diff
@@ -386,16 +383,57 @@ def gammas(self) -> List[Callable[[float], float]]:
         """List of gammas. """
         return copy(self._gammas)
 
-    @property
-    def field_eom(self) -> Callable[[float, float, ndarray, complex], complex]:
-        """The field equation of motion. """
-        return copy(self._field_eom)
-
     @property
     def lindblad_operators(self) -> List[Callable[[float], ndarray]]:
         """List of lindblad operators. """
         return copy(self._lindblad_operators)
 
+class MeanFieldSystem(BaseAPIClass):
+    r"""Represents a collection of time dependent systems interacting
+    with a common field. The systems are encoded as
+    `TimeDependentSystemWithField` objects, and the field as a complex
+    scalar :math:`\langle a \rangle` that evolves according to a
+    specified equation of motion :math:`\partial_t\langle a \rangle`.
+
+    Parameters
+    ----------
+    system_list: List[TimeDependentSystemWithField],
+        List of `TimeDependentSystemWithField` objects interacting with
+        a common field :math:`\langle a \rangle`.
+    field_eom: callable
+        Field equation of motion :math:`\partial_t
+        \langle a \rangle(t, [\rho], \langle a \rangle)`
+        where :math:`[\rho]` is a list of square matrices for the state
+        of each system in `system_list` at time :math:`t` and
+        :math:`\langle a \rangle` the field at time :math:`t`.
+    name: str
+        An optional name for the mean-field system.
+    description: str
+        An optional description of the mean-field system.
+    """
+
+    def __init__(self,
+                system_list: List[TimeDependentSystemWithField],
+                field_eom: Callable[[float, List[ndarray], complex], complex],
+                name: Optional[Text] = None,
+                description: Optional[Text] = None) -> None:
+
+        super().__init__(name, description)
+        tmp_system_list = _check_mean_field_system_list(system_list)
+        self.system_list = tmp_system_list
```

Comment:
I wasn't sure whether the property definition should return a copy. I don't think it should, since it's a public attribute, compared to the `_field_eom` which is private (and so copied when accessed as `field_eom`). Useful to get some experience of when to use the Python descriptors